### ISSUE TITLE
Region specification and key vault purge protection

### ISSUE THREAD

**Issue by abdale** [Azure/data-product-streaming#33]
I tried to deploy the data domain in a different region as the data landing zone resulting in this error:

![image](https://user-images.githubusercontent.com/8771696/115612221-05823180-a2b9-11eb-8232-755ae6aa3538.png)

During the failed deployment, the key vault was created inside the domain rg. I had to delete the domain rg so I can re-create it under the same region as the data landing zone.

The deployment failed again this time with this error:

![image](https://user-images.githubusercontent.com/8771696/115612357-2c406800-a2b9-11eb-899e-216aad9bd396.png)

**Assuming I want to keep the same name for my domain rg**, it seems like I could not do that as the key vault is not purged. When I try to purge the keyvault from the portal it fails as during the creation process, purge protection was enabled. Even if I recover the key vault so I can disable purge protection, I am not able to disable it as once enabled, this option cannot be disabled.

See:

![image](https://user-images.githubusercontent.com/8771696/115613662-b1784c80-a2ba-11eb-9585-b1f74e46e657.png)

So basically I have no option but to change the name of my data domain and redeploy.

Two potential changes that could avoid this scenario from happening:

- Update the [prerequisites](https://github.com/Azure/data-domain-streaming#prerequisites) to say that the region of the data domain must be the same as the data landing zone instead of saying choose any region
- [Disable purge protection](https://github.com/Azure/data-domain-streaming/blob/3ff4438d96184d25e424a3b0a652f7869c2969a4/infra/KeyVault/deploy.keyVault.json#L72) on the key vault
